test/CrossImport/horrible.swift

```diff
@@ -0,0 +1,44 @@
+// RUN: %empty-directory(%t)
+// RUN: cp -r %S/Inputs/lib-templates/ %t/
+
+// RUN: %target-typecheck-verify-swift -enable-cross-import-overlays -I %t/lib/swift
+
+//
+// This module declares some really bad cross-import files.
+//
+import HorribleGoose
+
+//
+// Let's load them!
+//
+
+import UnitaryGoose
+// expected-error@-1 {{cannot list cross-import overlays for 'UnitaryGoose': Not a directory}}
+
+// FIXME: It might be better to diagnose these errors on HorribleGoose's import
+// decl, since they actually belong to HorribleGoose.
```

Comment:
Could you file a radar to improve the diagnostic location for this one? I think for the same file case, this is probably fine, but for the `@_exported import` case [when we add that], it might be a bit of a head-scratcher what is going on if you don't understand the name lookup there.

Reply:
The SourceLoc thing is actually kind of interesting. What I'm currently doing is diagnosing with the location of the `ImportDecl` that caused us to load the swiftoverlay file. This has the upside that it will always be in user code, even if it's been imported indirectly, and that commenting out that ImportDecl will always cause the error to either go away or move. But it has the downsides that it's not necessarily the import *most directly connected* to the overlay file, and that it's not necessarily clear why the compiler decided it needed that cross-import there.

I think it would be better to include notes pointing to all of the imports involved in a cross-import overlay (hopefully even using generated interfaces to point to `@_exported imports` in modules), but `ModuleDecl::ImportedModule` doesn't include the necessary information to do that, so I think it will be difficult. The current implementation reliably gives us a reasonably actionable source location without doing a lot of engineering.

But I can certainly file an SR if you'd like.

Reply:
> I think it would be better to include notes pointing to all of the imports involved in a cross-import overlay (hopefully even using generated interfaces to point to @_exported imports in modules), but ModuleDecl::ImportedModule doesn't include the necessary information to do that, so I think it will be difficult. The current implementation reliably gives us a reasonably actionable source location without doing a lot of engineering.
>
> But I can certainly file an SR if you'd like.

I agree having that would be ideal. I think filing an SR would be useful even if we are not implementing this right away. It indicates that this is a known problem, in case someone runs into it.

Reply:
https://bugs.swift.org/browse/SR-12223

I'll add that number to the comment, too.

Reply:
Done.